dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -403,14 +403,31 @@ await foreach (Completions completions in response.ConfigureAwait(false))
                 }
 
                 // Now, invoke the function, and add the resulting tool call message to the chat options.
+                FunctionResult functionResult = new(function) { Culture = kernel.Culture };
```

Comment:
This will allow us to keep `FunctionResult Result` property in context as non-nullable. During my testing I realized that when `Result` property is non-nullable it's easier to use it and avoid null checks every time. Before `await next(context)` call, `Result` will have `null` value and after `await next(context)` call it will contain the actual value that function returned.